src/components/Buttons.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from '../components/Buttons.scss';
+
+const ButtonCreator = (type) => {
+  const Button = ({ onClick, children, tabIndex, ...props }) => (
+    <button
+      className={type}
+      tabIndex={tabIndex}
+      {...props}
+      onClick={() => onClick()}
+    >
+      {children}
+    </button>);
+
+  Button.propTypes = {
+    onClick: PropTypes.func.isRequired,
+    children: PropTypes.node.isRequired,
+    tabIndex: PropTypes.string.isRequired,
+    button: PropTypes.string.isRequired,
+  };
+
+  Button.defaultProps = {
+    tabIndex: '0',
+  };
+  return Button;
+};
```

Comment:
This can be inlined even more:

```
const ButtonCreator = (type) => {
  const Button = …;
  return Button;
}
```

can be simplified to

```
const ButtonCreator = (type) => …;
```